src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,10 +1,82 @@
-import React from 'react';
+import classNames from 'classnames';
+import React, { Dispatch, SetStateAction, useState } from 'react';
+import { getMovie } from '../../api';
+import { Movie } from '../../types/Movie';
+import { MovieData } from '../../types/MovieData';
+import { ResponseError } from '../../types/ReponseError';
+import { MovieCard } from '../MovieCard';
 import './FindMovie.scss';
 
-export const FindMovie: React.FC = () => {
+type Props = {
+  onAddMovies: Movie[]
+  setMovies: Dispatch<SetStateAction<Movie[]>>
+};
+
+export const FindMovie: React.FC<Props> = ({ onAddMovies, setMovies }) => {
+  const [query, setQuery] = useState<string>('');
+  const [movie, setMovie] = useState<Movie | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState(false);
+
+  function isError(a: MovieData | ResponseError): a is ResponseError {
+    return (a as ResponseError).Error !== undefined;
+  }
+
+  const normalizedMovie = (data: MovieData): Movie => {
+    return (
+      {
+        title: data.Title,
+        description: data.Plot,
+        imgUrl: data.Poster !== 'N/A'
+          ? data.Poster
+          : 'https://via.placeholder.com/360x270.png?text=no%20preview',
+        imdbUrl: `https://www.imdb.com/title/${data.imdbID}`,
+        imdbId: data.imdbID,
+      }
+    );
+  };
+
+  const fetchData = async () => {
+    setIsLoading(true);
+    const data = await getMovie(query);
+    // const x = isError(data);
+
+    // console.log(query);
+    // console.log('TRUE OR FALSE with isError', x);
+    setErrorMessage(isError(data));
+    if (!isError(data)) {
+      setMovie(normalizedMovie(data));
+    }
+
+    setIsLoading(false);
+  };
+
+  const handleQuery = (e: string) => {
+    setErrorMessage(false);
+    setQuery(e);
+  };
+
+  const handleFindMovie = () => {
+    fetchData();
+  };
+
+  const handleAddToList = () => {
+    setQuery('');
+
+    const isMovieInList = onAddMovies
+      .find(existMovie => existMovie.imdbId === movie?.imdbId);
+
+    setMovies(movie && !isMovieInList
+      ? (a: Movie[]) => [...a, movie] : onAddMovies);
+
```

Comment:
Move your condition in upper scope, so it will be more understandable what does it mean. And also don't name your variable `a`
```suggestion
if (movie && !isMovieInList) {
    setMovies(prevMovies => [...prevMovies, movie]);
}
```